AEPAssurance/Source/AssuranceSessionOrchestrator.swift

```diff
@@ -24,68 +40,114 @@ class AssuranceSessionOrchestrator: AssurancePresentationDelegate {
 
     init(stateManager: AssuranceStateManager) {
         self.stateManager = stateManager
+        outboundEventBuffer = ThreadSafeArray(identifier: "Session Orchestrator's OutboundBuffer array")
     }
 
-    func createSession() {
+    /// Creates and starts a new  `AssuranceSession` with the provided `AssuranceSessionDetails`.
+    ///
+    /// A new AssuranceSession is only created if there isn't an already existing session.
+    /// Calling this method also shares the shared state for the extension with the provided session details
+    ///
+    /// - Parameters:
+    ///    - sessionDetails: An `AssuranceSessionDetails` instance containing all the essential data for starting a session
+    func createSession(withDetails sessionDetails: AssuranceSessionDetails) {
+        if let _ = session {
+            Log.warning(label: AssuranceConstants.LOG_TAG, "An active session already exists. Cannot create a new one.")
+            return
+        }
+
+        stateManager.shareAssuranceState(withSessionID: sessionDetails.sessionId)
+        session = AssuranceSession(sessionDetails: sessionDetails, stateManager: stateManager, sessionOrchestrator: self, outboundEvents: outboundEventBuffer)
+        session?.startSession()
 
+        outboundEventBuffer?.clear()
+        outboundEventBuffer = nil
     }
 
+    ///
+    /// Dissolve the active session (if one exists) and its associated states.
+    ///
     func terminateSession() {
+        hasEverTerminated = true
+        self.outboundEventBuffer = nil
+
+        stateManager.clearAssuranceState()
 
+        session?.disconnect()
+        session = nil
     }
 
-    func shutDownSession() {
+    func queueEvent(_ assuranceEvent: AssuranceEvent) {
+        /// Queue this event to the active session if one exists.
+        if let session = session {
```

Comment:
nit: code on line 54 uses if let _ = session, where as this uses the same-name if let pattern.  Either are fine with me, but I'd prefer they be consistent (and I think most other places use the same-name pattern).

Reply:
I think I am using if let the wrong way
all the empty  `if let _ = xyz { } ` are usually only to check if the variable is not nil... hence replaces it with 
`if xyx != nil {}`  (they don't need an assignment)

Other places I will probably keep the same-name pattern where the unwrapped variable is used 

```
        if let session = session {
            session.sendEvent(assuranceEvent)
            return
        }
```